### PR TITLE
Add missing shader debug documentation

### PR DESCRIPTION
**Pull request by AThousandShips**
Introduced by #73588, fixes CI

### CODE REVIEW

**Comment by AThousandShips** [comment]
Filling in these descriptions

File: doc/classes/ProjectSettings.xml [hunk 1]
```diff
@@ -546,6 +546,36 @@
 		<member name="debug/settings/stdout/verbose_stdout" type="bool" setter="" getter="" default="false">
 			Print more information to standard output when running. It displays information such as memory leaks, which scenes and resources are being loaded, etc. This can also be enabled using the [code]--verbose[/code] or [code]-v[/code] [url=$DOCS_URL/tutorials/editor/command_line_tutorial.html]command line argument[/url], even on an exported project. See also [method OS.is_stdout_verbose] and [method @GlobalScope.print_verbose].
 		</member>
+		<member name="debug/shader_language/warnings/device_limit_exceeded" type="bool" setter="" getter="" default="true">
+		</member>
+		<member name="debug/shader_language/warnings/enable" type="bool" setter="" getter="" default="true">
+			If [code]true[/code], enables specific shader warnings (see [code]debug/shader_language/warnings/*[/code] settings). If [code]false[/code], disables all shader warnings.
+		</member>
+		<member name="debug/shader_language/warnings/float_comparison" type="bool" setter="" getter="" default="true">
```
Comment (clayjohn) [inline doc/classes/ProjectSettings.xml:554]:
```suggestion
		<member name="debug/shader_language/warnings/float_comparison" type="bool" setter="" getter="" default="true">
				When set to [code]true[/code], produces a warning when two floating point numbers are compared directly with the [code]==[/code] operator or the [code]!=[/code] operator.	
```

File: doc/classes/ProjectSettings.xml [hunk 2]
```diff
@@ -546,6 +546,36 @@
 		<member name="debug/settings/stdout/verbose_stdout" type="bool" setter="" getter="" default="false">
 			Print more information to standard output when running. It displays information such as memory leaks, which scenes and resources are being loaded, etc. This can also be enabled using the [code]--verbose[/code] or [code]-v[/code] [url=$DOCS_URL/tutorials/editor/command_line_tutorial.html]command line argument[/url], even on an exported project. See also [method OS.is_stdout_verbose] and [method @GlobalScope.print_verbose].
 		</member>
+		<member name="debug/shader_language/warnings/device_limit_exceeded" type="bool" setter="" getter="" default="true">
```
Comment (clayjohn) [inline doc/classes/ProjectSettings.xml:549]:
```suggestion
		<member name="debug/shader_language/warnings/device_limit_exceeded" type="bool" setter="" getter="" default="true">
			When set to [code]true[/code], produces a warning when the shader exceeds certain device limits. Currently, the only device limit checked is the limit on uniform buffer size. More device limits will be added in the future.
```

File: doc/classes/ProjectSettings.xml [hunk 3]
```diff
@@ -546,6 +546,36 @@
...
+		</member>
+		<member name="debug/shader_language/warnings/enable" type="bool" setter="" getter="" default="true">
+			If [code]true[/code], enables specific shader warnings (see [code]debug/shader_language/warnings/*[/code] settings). If [code]false[/code], disables all shader warnings.
+		</member>
+		<member name="debug/shader_language/warnings/float_comparison" type="bool" setter="" getter="" default="true">
+		</member>
+		<member name="debug/shader_language/warnings/formatting_error" type="bool" setter="" getter="" default="true">
```
Comment (clayjohn) [inline doc/classes/ProjectSettings.xml:556]:
```suggestion
		<member name="debug/shader_language/warnings/formatting_error" type="bool" setter="" getter="" default="true">
			When set to [code]true[/code], produces a warning upon encountering certain formatting errors. Currently this only checks for empty statements. More formatting errors may be added over time.
```

**Review by clayjohn — APPROVED** [review]
Looks great! I hope I didn't step on your toes, I started suggesting descriptions before your comment about filling in the descriptions haha!

**Comment by AThousandShips** [comment]
Oh no worries! I didn't quite know what to fill into those and very appreciated

**Comment by AThousandShips** [comment]
Everything should be ready

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank you!

**Pull request merged**